pkg/types/unix_milli.go

```diff
@@ -30,32 +29,32 @@ func (t UnixMilli) MarshalJSON() ([]byte, error) {
 	return []byte(strconv.FormatInt(time.Time(t).UnixMilli(), 10)), nil
 }
 
-// UnmarshalText implements the encoding.TextUnmarshaler interface.
-func (t *UnixMilli) UnmarshalText(text []byte) error {
-	parsed, err := strconv.ParseFloat(string(text), 64)
-	if err != nil {
-		return internal.CantParseFloat64(err, string(text))
-	}
-
-	*t = UnixMilli(utils.FromUnixMilli(int64(parsed)))
-	return nil
-}
-
 // UnmarshalJSON implements the json.Unmarshaler interface.
 // Unmarshals from milliseconds. Supports JSON null.
 func (t *UnixMilli) UnmarshalJSON(data []byte) error {
-	if string(data) == "null" || len(data) == 0 {
+	if bytes.Equal(data, []byte("null")) || len(data) == 0 {
 		return nil
 	}
 
-	ms, err := strconv.ParseFloat(string(data), 64)
-	if err != nil {
-		return internal.CantParseFloat64(err, string(data))
+	return t.fromByteString(data)
+}
+
+// MarshalText implements the encoding.TextMarshaler interface.
+func (t UnixMilli) MarshalText() ([]byte, error) {
+	if time.Time(t).IsZero() {
+		return []byte{}, nil
 	}
-	tt := utils.FromUnixMilli(int64(ms))
-	*t = UnixMilli(tt)
 
-	return nil
+	return []byte(strconv.FormatInt(time.Time(t).UnixMilli(), 10)), nil
```

Comment:
Isn't the `Time()` method there to make this cast more convenient, i.e. `t.Time()` instead of `time.Time(t)`?